.github/workflows/ci.yml

```diff
@@ -16,7 +16,55 @@ permissions:
   contents: read # to fetch code (actions/checkout)
 
 jobs:
+  cs-stan:
+    name: Coding Standard & Static Analysis
+    runs-on: ubuntu-22.04
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '7.4'
+        extensions: mbstring, intl, apcu, memcached, redis
+        tools: cs2pr
+        coverage: none
+
+    - name: Get composer cache directory
+      id: composer-cache
+      run: echo "dir=$(composer config cache-files-dir)" >> $GITHUB_OUTPUT
+
+    - name: Get date part for cache key
+      id: key-date
+      run: echo "date=$(date +'%Y-%m')" >> $GITHUB_OUTPUT
+
+    - name: Cache composer dependencies
+      uses: actions/cache@v3
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ steps.key-date.outputs.date }}-${{ hashFiles('composer.json') }}-${{ matrix.prefer-lowest }}
+
+    - name: Composer install
+      run: composer stan-setup
+
+    - name: Run PHP CodeSniffer
+      run: vendor/bin/phpcs --report=checkstyle src/ tests/ | cs2pr
+
+    - name: Run psalm
+      if: always()
+      run: vendor/bin/psalm.phar --output-format=github
+
+    - name: Run phpstan
+      if: always()
+      run: vendor/bin/phpstan.phar analyse --error-format=github
+
+    - name: Run phpstan for tests
+      if: always()
+      run: vendor/bin/phpstan.phar analyse -c tests/phpstan.neon --error-format=github
+
   testsuite:
+    needs: cs-stan
```

Comment:
Why did this need to change?

Reply:
I noticed when I submitted this pull request that all of the jobs ran in parallel. So it spun up the different environments for testing along with the job for CS and Stan checks. I made this change so that the CS and Stan checks would run, and only kick off the test suites is the Checks passed first. This is to keep the test suites from running if the only thing that needs to be fixed is something like the Use [classname]; list is out of order or something.
It basically just saves resources, especially if you pay for said resources to run. It saves those cycles.
My CI runs like:
phpcs -> phpstan -> phpunit

Reply:
Please revert this revert this, we don't want to block the running of test suite in case of CS/stan failures. Getting info about failed tests earlier is more important for us.